prov/sockets/src/sock_ep_msg.c

```diff
@@ -822,124 +565,145 @@ static int sock_ep_cm_connect(struct fid_ep *ep, const void *addr,
 	if (!_ep->attr->listener.listener_thread && sock_conn_listen(_ep->attr))
 		return -FI_EINVAL;
 
-	req = calloc(1, sizeof(*req) + paramlen);
+	if (!_ep->attr->dest_addr) {
+		_ep->attr->dest_addr = calloc(1, sizeof(*_ep->attr->dest_addr));
+		if (!_ep->attr->dest_addr)
+			return -FI_ENOMEM;
+		memcpy(_ep->attr->dest_addr, addr, sizeof(*_ep->attr->dest_addr));
```

Comment:
This line should be moved out of <code>if (!_ep->attr->dest_addr)</code>. We need to update if <code>_ep->attr->dest_addr</code> regardless the check.